lib/ex_teal_direct_upload/aws_sts.ex

```diff
@@ -0,0 +1,41 @@
+defmodule ExTealDirectUpload.AwsSts do
+  @moduledoc """
+  Module for handling AWS STS
+  for temporary assignment of roles to upload users
+  """
+
+  @session_name "ex_teal_direct_upload"
+  @credentials_duration 3_600
+
+  def config do
+    case assume_role() do
+      {:ok, %{body: credentials}} ->
+        ExAws.Config.new(:s3,
+          access_key_id: credentials.access_key_id,
+          secret_access_key: credentials.secret_access_key,
+          security_token: credentials.session_token,
+          region: region()
+        )
+
+      _ ->
+        {:error, :invalid_role_request}
+    end
+  end
+
+  def enabled? do
+    case role_arn() do
```

Comment:
I'd maybe suggest here that we also check for `Code.ensure_loaded?(ExAws.STS)` so that we can make `ex_aws_sts` an optional dependency